instance/tests/models/test_server.py

```diff
@@ -129,21 +130,28 @@ def test_sleep_until_status(self, mock_sleep, mock_update_status):
         Sleep until the server gets to 'booted' status (single status string argument)
         """
         server = OpenStackServerFactory()
-        status_queue = [server.STARTED, server.STARTED, server.ACTIVE, server.BOOTED, server.TERMINATED]
+        status_queue = [
+            None,
+            server._status_to_started,
+            server._status_to_active,
+            server._status_to_booted,
+            server._status_to_terminated,
+        ]
         status_queue.reverse() # To be able to use pop()
 
         def update_status():
             """ Simulate status progression successive runs """
-            server.status = status_queue.pop()
-            server.progress = server.PROGRESS_SUCCESS
+            new_status = status_queue.pop()
+            if new_status:
+                server._transition(new_status, progress=ServerProgress.Success)
         mock_update_status.side_effect = update_status
 
-        self.assertEqual(server.sleep_until_status(server.BOOTED), server.BOOTED)
-        self.assertEqual(server.progress, server.PROGRESS_SUCCESS)
-        self.assertEqual(server.status, server.BOOTED)
-        self.assertEqual(server.progress, server.PROGRESS_SUCCESS)
+        self.assertEqual(server.sleep_until_status(ServerStatus.Booted), ServerStatus.Booted)
+        self.assertEqual(server.progress, ServerProgress.Success)
+        self.assertEqual(server.status, ServerStatus.Booted)
+        self.assertEqual(server.progress, ServerProgress.Success)
```

Comment:
@bradenmacdonald nit: double assertion on server.progress - here and two lines above